fix: disallow child unignoring parent directory ignore

mohd-akram opened the pull request:
This solves issues with `npm pack` (this library is used by [npm-packlist](https://github.com/npm/npm-packlist) which implements the command).

## References
Related to npm/cli#7007

wraithgar commented:
Perhaps I'm not seeing it clearly enough, but I don't see how this distinguishes between a file that was excluded in the parent which is now being included, and a file that is in a directory that was excluded in the parent which is now being included.

The former should still work, if I read the spec correctly.

mohd-akram commented:
You're right. In my testing I thought git was ignoring both categories but it seems I mis-tested. I've adjusted the PR and added new tests. The added test fails on the `main` branch as expected.

mohd-akram commented:
Makes sense. I've updated the PR.

wraithgar commented:
Great! Almost there.  It looks like the test includes "directories that are ignored can't have files in it be unignored" but does it have "files that are ignored can be unignored"?

mohd-akram commented:
@wraithgar Those are tested in `test/nested-ignores.js` already.

wraithgar commented:
This closed #17 I believe.  If not we can reopen it.
